main.ts

```diff
@@ -133,10 +158,33 @@ class BeautitabPluginSettingTab extends PluginSettingTab {
 		this.plugin = plugin;
 	}
 
+	/**
+	 * Filter SearchProvider based on what plugins are installed
+	 */
+	getSearchProvider() {
+		const app = this.plugin.app;
+		//@ts-ignore
+		const plugins = app.plugins.enabledPlugins as Set<string>;
+
+		const searchProviders = SearchProviders.filter((provider) => {
+			return plugins.has(provider.value.split(":")[0]);
```

Comment:
Instead of the `push` below and replicating the entry, update this `filter` with an `||`. Can also update this to be a 1 liner arrow function without the `return`.

```javascript
const searchProviders = SearchProviders.filter((provider) => 
			plugins.has(provider.value.split(":")[0]) || provider.value === SearchProvider.SWITCHER
);
```

That being said, I'm not sure I want to completely hide the options as it sort of hides functionality.

My preference would be to add the options to the dropdown in settings but disable the ones where the plugin isn't installed. Additionally, I would update the setting description to mention that only installed plugins can be selected from the list.

So probably instead of having this function, I would check if a plugin is installed while adding it to the dropdown and if it is not, disable the option.

Reply:
OH darn, I just checked and it doesn't appear that you can disable individual options in Obsidian. :(

Instead then, when the user selects an option, check if the plugin is enabled... if it is, cool, if it isn't, throw a Notice that the plugin is not enabled and revert back to the previous option (call `display` to reset).